lib/puppet/type/wls_datasource/connectionreservetimeoutseconds.rb

```diff
@@ -0,0 +1,12 @@
+newproperty(:connectionreservetimeoutseconds) do
+  include EasyType
+
+  desc 'The number of seconds after which a call to reserve a connection from the connection pool will timeout.'
+
+  defaultto '10'
```

Comment:
indeed,

but with the new set_attribute_value('FrontendHTTPSPort', frontendhttpsport) I will retrieve the mbean attribute defaults if the value is not provided and use that value.

So if you set a property value and after that you remove this property from the wls type definition I will set it back to the default by the next change. Model driven and the puppet configuration is always leading, No manual change is allowed.

I was planning to retrieve all the attributes defaults during the index phase and use that. This will work on all wls versions. But I need to think a little bit  more about this.